FFDH 4: driver-only parity testing - with TLS 1.3

mprse opened the pull request:
## Description

Resolves: https://github.com/Mbed-TLS/mbedtls/issues/7569

Needs: https://github.com/Mbed-TLS/mbedtls/pull/7577 https://github.com/Mbed-TLS/mbedtls/pull/7627

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [X] **changelog** not required (just test component)
- [X] **backport** not required
- [X] **tests** provided

mpg reviewed (commented):
(Just a first look.)

mprse commented on tests/scripts/all.sh:
```diff
@@ -2204,6 +2207,34 @@ component_test_psa_crypto_config_accel_ffdh () {
+    # Disable things that are not supported
+    scripts/config.py unset MBEDTLS_KEY_EXCHANGE_DHE_PSK_ENABLED
+    scripts/config.py unset MBEDTLS_KEY_EXCHANGE_DHE_RSA_ENABLED
+    scripts/config.py unset MBEDTLS_DHM_C
```
Fixed

mprse commented:
Rebased this PR and added some adaptations in [commit](https://github.com/Mbed-TLS/mbedtls/pull/7656/commits/19a744721b118d1e956f013d4fa03cebad740eb5).

mpg commented on tests/scripts/all.sh:
```diff
@@ -2261,7 +2261,7 @@ component_test_psa_crypto_config_accel_ffdh () {
 
     # Start from default config (no TLS 1.3, no USE_PSA)
-    helper_libtestdriver1_adjust_config "default"
+    helper_libtestdriver1_adjust_config "full"
```
Should update the comment above too :)

mpg commented on tests/scripts/all.sh:
```diff
@@ -2285,6 +2285,37 @@ component_test_psa_crypto_config_accel_ffdh () {
+    msg "build: MBEDTLS_PSA_CRYPTO_CONFIG with accelerated FFDH"
+
+    # Start with full (USE_PSA and TLS 1.3)
+    scripts/config.py full
```
Suggestion: use `helper_libtestdriver1_adjust_config "full"`, then most lines below become redundant (only need to keep the ones about `MBEDTLS_KEY_EXCHANGE_DHE_xxx_ENABLED` I think). This keeps the reference component closer to the accel component, too.

mpg commented on tests/scripts/analyze_outcomes.py:
```diff
@@ -328,6 +328,15 @@ def do_analyze_driver_vs_reference(outcome_file, args):
+            'component_ref': 'test_psa_crypto_config_reference_ffdh',
+            'component_driver': 'test_psa_crypto_config_accel_ffdh',
+            'ignored_suites': ['dhm'],
+            'ignored_tests': {'test_suite_pkparse': ['DH group family: RFC 7919']}
```
I'm not seeing any such test in `test_suite_pkparse`, what am I missing?

mprse replied:
Oh ok. This can be removed, because I changed the idea to adapt test dependencies instead of removing this test case and left this by mistake (also invalid version with bad test name):

```
/* BEGIN_CASE depends_on:PSA_KEY_TYPE_DH_PUBLIC_KEY:PSA_KEY_TYPE_DH_KEY_PAIR */
void dh_key_family(int group_arg)
```

mpg reviewed (commented):
Looking pretty good, just a few suggestions.

mpg reviewed (commented):
LGTM, thanks!

valeriosetti reviewed (approved):
LGTM ;)
